### PR TITLE
Plfm 5795 b

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/form/FormManagerImpl.java [hunk 1]
```diff
@@ -24,21 +27,25 @@
...
 import org.sagebionetworks.repo.model.form.ListResponse;
 import org.sagebionetworks.repo.model.form.StateEnum;
+import org.sagebionetworks.repo.model.form.SubmissionStatus;
 import org.sagebionetworks.repo.model.util.AccessControlListUtil;
 import org.sagebionetworks.repo.transactions.WriteTransaction;
 import org.sagebionetworks.util.ValidateArgument;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import com.amazonaws.services.pinpointsmsvoice.model.NotFoundException;
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/form/FormManagerImpl.java]: wrong not found

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/form/FormManagerImpl.java [hunk 2]
```diff
@@ -171,69 +186,197 @@ static void validateCanUpdate(StateEnum currentState) {
...
+		response.setPage(page);
+		response.setNextPageToken(token.getNextPageTokenForCurrentResults(page));
+		return response;
 	}
 
 	@Override
-	public FormData reviewerAcceptForm(UserInfo user, String id) {
-		// TODO Auto-generated method stub
-		return null;
+	public FormData reviewerAcceptForm(UserInfo user, String formDataId) {
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/form/FormManagerImpl.java:340]: transaction

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/form/FormManagerImpl.java [hunk 3]
```diff
@@ -171,69 +186,197 @@ static void validateCanUpdate(StateEnum currentState) {
...
+		ValidateArgument.required(user, "UserInfo");
+		ValidateArgument.required(formDataId, "formDataId");
+
+		// must have the READ_PRIVATE_SUBMISSION permission on the group.
+		validateGroupPermission(user, formDataId, ACCESS_TYPE.READ_PRIVATE_SUBMISSION);
+
+		SubmissionStatus status = formDao.getFormDataStatus(formDataId);
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/form/FormManagerImpl.java]: get for update

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/form/FormManagerImpl.java [hunk 4]
```diff
@@ -171,69 +186,197 @@ static void validateCanUpdate(StateEnum currentState) {
...
+		status.setReviewedOn(new Date(System.currentTimeMillis()));
+		status.setState(StateEnum.ACCEPTED);
+		return formDao.updateStatus(formDataId, status);
 	}
 
 	@Override
-	public FormData reviewerRejectForm(UserInfo user, String reason) {
-		// TODO Auto-generated method stub
-		return null;
+	public FormData reviewerRejectForm(UserInfo user, String formDataId, String reason) {
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/form/FormManagerImpl.java:360]: transaction

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/form/FormDao.java [hunk 5]
```diff
@@ -27,61 +30,115 @@
...
 	 * Truncate all data.
 	 */
 	public void truncateAll();
+
+	/**
+	 * Delete the identified FormData
+	 * 
+	 * @param formDataId
+	 */
+	public boolean deleteFormData(String formDataId);
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/form/FormDao.java:112]: document boolean

File: lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/form/FormDaoImpl.java [hunk 6]
```diff
@@ -206,10 +225,113 @@ static FormData dtoToDbo(DBOFormData dbo) {
...
+		try {
+			return StateEnum.valueOf(jdbcTemplate.queryForObject("SELECT " + COL_FORM_DATA_STATE + " FROM "
+					+ TABLE_FORM_DATA + " WHERE " + COL_FORM_DATA_ID + " = ?", String.class, formDataId));
+		} catch (EmptyResultDataAccessException e) {
+			throw new NotFoundException(String.format(FORM_DATA_DOES_NOT_EXIST_FOR_S, formDataId));
+		}
+	}
+
+	@Override
+	public SubmissionStatus getFormDataStatus(String formDataId) {
```
Comment (john-hill) [inline lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/form/FormDaoImpl.java]: for update

**Review by marcomarasca — APPROVED** [review]

**Pull request merged**